ranndy/models.py

```diff
@@ -18,16 +21,20 @@ def __init__(self, data_iterator, mode=tf.estimator.ModeKeys.TRAIN):
         self.mode = mode
 
         # TODO: set up hyperparams properly
-        self.embedding_size = 256
+        self.embedding_size = 512
         self.lstm_size = 512
         self.batch_size = data_iterator.batch_size
-        self.num_epochs = 10
+        self.max_num_epochs = 15
         self.max_gradient_norm = 1.
         self.learning_rate = 0.0001
 
         # Data iterator
         self.iterator = data_iterator
 
+        # Convergence parameters
+        self.conv_window_size = 4  # Number of epochs to consider when checking for
+        self.conv_loss_threshold = 100000 # Minimum loss decrease required to continue
```

Comment:
Where did you get 10000 from?

Reply:
Was using a big number to prevent early convergence. I'll reduce it to something more reasonable.